File: HLRouter/HLRouter/ViewController.m
//
//  ViewController.m
//  HLRouter
//
//  Created by HomerLynn on 2017/11/21.
//  Copyright © 2017年 HomerLynn. All rights reserved.
//

#import "ViewController.h"
#import "HLRouter.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.

}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event {
    NSString *url = @"https://SecondViewController?paramOne=hello&person={\"name\":\"xiaoming\",\"age\":\"18\"}";
    [[HLRouter share] pushWith:url];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Router/Router/ViewController.swift
//
//  ViewController.swift
//  Router
//
//  Created by HomerLynn on 2017/11/3.
//  Copyright © 2017年 HomerLynn. All rights reserved.
//

import UIKit

class ViewController: UIViewController, RouteAble {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        let url = "https://SecondViewController?hello=123&world=234"
        let secondVC = SecondViewController()
        secondVC.url = url
        navigationController?.pushViewController(secondVC, animated: true)
    }
    
}
